src/app/my-coupons/page.tsx

```diff
@@ -0,0 +1,203 @@
+    icon: CouponRed,
+  },
+];
+
+const ArrayCoupons = [
+  {
+    id: 12312312312,
+    status: "active",
+    title: "Creatina 40%",
+    discount: 40,
+    maxTotal: 20,
+    maxPerUser: 5,
+    expirationGenerationDate: "2024-01-01",
+    expirationUseDate: "2024-02-02",
+  },
+  {
+    id: 12312312312,
+    status: "active",
+    title: "Whey 20%",
+    discount: 20,
+    maxTotal: 20,
+    maxPerUser: 5,
+    expirationGenerationDate: "2024-01-01",
+    expirationUseDate: "2024-02-02",
+  },
+  {
+    id: 2312312,
+    status: "used",
+    usageDate: "2023-10-02",
+    title: "Creatina 40%",
+    discount: 40,
+    maxTotal: 20,
+    maxPerUser: 5,
+    expirationGenerationDate: "2024-01-01",
+    expirationUseDate: "2024-02-02",
+  },
+  {
+    id: 1231234124312,
+    status: "used",
+    usageDate: "2023-10-02",
+    title: "Whey 20%",
+    discount: 20,
+    maxTotal: 20,
+    maxPerUser: 5,
+    expirationGenerationDate: "2024-01-01",
+    expirationUseDate: "2024-02-02",
+  },
+  {
+    id: 223124123412312,
+    status: "expired",
+    title: "Creatina 40%",
+    discount: 40,
+    maxTotal: 20,
+    maxPerUser: 5,
+    expirationGenerationDate: "2024-01-01",
+    expirationUseDate: "2024-02-02",
+  },
+  {
+    id: 123141123124312,
+    status: "expired",
+    title: "Whey 20%",
+    discount: 20,
+    maxTotal: 20,
+    maxPerUser: 5,
+    expirationGenerationDate: "2024-01-01",
+    expirationUseDate: "2024-02-02",
+  },
+];
+
+const MyCouponsPage = () => {
+  const { data: session } = useSession();
+  const [couponsFilter, setCouponsFilter] = useState<filterOptions>(
+    filterOptions[0],
+  );
+  const [coupons, setCoupons] = useState<ICouponsByUser>();
+
+  const handleGetCouponsByUser = async () => {
+    //TODO - API request to Get user coupons
+    const res: any = await ArrayCoupons;
+    return res;
+  };
+
+  useEffect(() => {
+    handleGetCouponsByUser()
+      .then((res) => setCoupons(res))
+      .catch((error) =>
+        showToastify({ type: "error", label: `Ocorreu um erro: ${error}` }),
+      );
+  }, [couponsFilter]);
+
+  return (
+    <div className="relative md:w-[500px] h-full w-full mx-auto">
+      <FloatButtonNav
+        href="/"
+        icon={CouponPrimary}
+        backgroundStyle="secondary"
+      />
+      <div className="mt-8 mb-16 flex items-center justify-between">
+        <h2 className=" pl-6 flex items-center text-2xl leading-[115%] md:leading-[115%] font-bold text-black dark:text-neutral-100 justify-center">
+          {session?.user.name}
+        </h2>
+        <div>
+          <div className="relative rounded-full w-40 h-16 bg-gradient-to-r from-secondary-dark to-secondary-lighter">
+            <div className="absolute top-8 right-0 rounded-full w-16 h-16 bg-gradient-to-r from-secondary-dark to-secondary-lighter"></div>
+          </div>
+        </div>
+      </div>
+      <div className="flex justify-end items-center w-full gap-4">
+        <div
+          className={`${
```

Comment:
Refactor for a ObjectMap.

const COUPONS_FILTER={
user: "text-black",
....
}